### PR TITLE
misc: drop support for Node before 10.13 (LTS)

### LINKED ISSUE #7215: Move to Node 10
It's [that](https://github.com/GoogleChrome/lighthouse/issues/449) time [again](https://github.com/GoogleChrome/lighthouse/issues/3742).

Node 10.13 was moved to LTS at the end of October 2018. Historically we've waited until some time in the new year to move over to the new version. [Last time we did it](https://github.com/GoogleChrome/lighthouse/pull/4703) with the split to 3.0.

There's not a ton of new stuff (for-await (including for readable streams), `Promise.finally`, BigInt, `fs/promises`, global `URL`, `--experimental-worker`), but they offer some pretty decent QoL improvements.

Regardless of timeline, we'll need to update our testing configs to start testing node 11 as well.

### PR DESCRIPTION
**Pull request by brendankenny**
fixes #7215 and part of #7752

### CODE REVIEW

File: lighthouse-core/lib/asset-saver.js [hunk 1]
```diff
@@ -139,9 +139,8 @@ async function prepareAssets(artifacts, audits) {
 }
 
 /**
- * Generates a JSON representation of traceData line-by-line to avoid OOM due to very large traces.
- * COMPAT: As of Node 9, JSON.parse/stringify can handle 256MB+ strings. Once we drop support for
- * Node 8, we can 'revert' PR #2593. See https://stackoverflow.com/a/47781288/89484
```
Comment (brendankenny) [inline lighthouse-core/lib/asset-saver.js]: even though we *can* do this now, I think we like the one trace event per line format, so it made sense to drop the old comment.

File: .appveyor.yml [hunk 2]
```diff
@@ -9,11 +9,11 @@ branches:
 environment:
   fast_finish: true
   matrix:
-    - nodejs_version: "8"
+    - nodejs_version: "10"
       platform: x86
     # other Node versions are skipped, as appveyor only allows 1 concurrent job
     # and we want appveyor finishing ASAP. see #2382
-    # - nodejs_version: "9"
+    # - nodejs_version: "11"
```
Comment (patrickhulce) [inline .appveyor.yml]: should we even bother to keep this in here?

**Comment by patrickhulce** [comment]
Hang on though, tsc isn't happy with new node types

File: .appveyor.yml (hunk 2 above)
Comment (brendankenny) [reply .appveyor.yml]: > should we even bother to keep this in here?

lol, no :)

**Comment by brendankenny** [comment]
there was a conflict with the new node types assuming that there would always be a `Error.code` (as there usually is in node). Fortunately it's easy to drop our `Error` augmentation since we've mostly switched over to `LHError`.

The last holdout fortunately fixes a long-time TODO (that we probably would have missed again for the major version change :), but on the downside is more annoying to review :)

File: lighthouse-core/lib/asset-saver.js (hunk 1 above)
Comment (Hoten) [reply lighthouse-core/lib/asset-saver.js]: yes, we love it

File: types/externs.d.ts [hunk 3]
```diff
@@ -202,10 +196,6 @@ declare global {
       group: string;
     }
 
-    export interface LighthouseError extends Error {
```
Comment (patrickhulce) [inline types/externs.d.ts]: might have been nice to just export our class as this type so we could easily reference it as `LH.LighthouseError` still but I don't feel strongly

Comment (brendankenny) [reply types/externs.d.ts]: yeah, I'd like to add this to our public types (when those actually exist :)

**Pull request merged**